go-controller/pkg/util/multi_network.go

```diff
@@ -417,50 +417,53 @@ func ParseNetConf(netattachdef *nettypes.NetworkAttachmentDefinition) (*ovncnity
 	return netconf, nil
 }
 
-// PodWantsMultiNetwork sees if the given pod needs to plumb over this given network specified by netconf,
+// PodWantsNetwork sees if the given pod needs to plumb over this given network specified by netconf,
 // and return the matching NetworkSelectionElement if any exists.
 //
 // Return value:
 //    bool: if this Pod is on this Network; true or false
-//    *nettypes.NetworkSelectionElement: all NetworkSelectionElement that pod is requested for the specified network
+//    map[string]*nettypes.NetworkSelectionElement: all NetworkSelectionElement that pod is requested
+//        for the specified network, key is NADName. Note multiple NADs of the same network are allowed
+//        on one pod, as long as they are of different NADName.
 //    error:  error in case of failure
-func PodWantsMultiNetwork(pod *kapi.Pod, nInfo NetInfo) (bool, *nettypes.NetworkSelectionElement, error) {
-	var validNADName string
+func PodWantsNetwork(pod *kapi.Pod, nInfo NetInfo) (bool, map[string]*nettypes.NetworkSelectionElement, error) {
+	if pod.Spec.HostNetwork {
+		return false, nil, nil
+	}
```

Comment:
Just had this thought. I guess that we are assuming that a host network pod cannot be attached to secondary networks. But API wise there is nothing preventing the user from attempting such a thing. Are we checking this somewhere and returning an error?

Reply:
I don't quite understand this comment. If a pod is host-networked, I think even multus-cni won't be triggered, as the result, multus won't be able to start cni for secondary network either.

Reply:
Yeah. So if a user attempts to run a pod with `hostNetwork: true` and annotated with `k8s.v1.cni.cncf.io/networks` (or `v1.multus-cni.io/default-network`) pointing to a NAD with an ovn-k secondary network, we simply ignore the annotation instead of reporting an error somewhere for that pod given that we cannot comply with what the user is requesting.